Improve: ensure whole item is selected in room properties QComboBoxes

SlySven opened the pull request:
By ensuring that the whole line of text is selected when the user selects an (editable) item in the Room Symbol or Weight `QComboBox` it means that the default action if they type a text entry key will be to delete the existing text (including any sort of "(count: X room(s)" suffix to the actual value). This UI improvement will make the entering a new value easier as they will not have to manually delete the existing entry first.

I believe this will close (the outstanding issue with) #6475.
 
Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Improve: ensure whole item is selected in room properties QComboBoxes

By ensuring that the whole line of text is selected when the user selects
an (editable) item in the Room Symbol or Weight `QComboBox` it means that
the default action if they type a text entry key will be to delete the
existing text (including any sort of "(count: X room(s)" suffix to the
actual value). This UI improvement will make the entering a new value
easier as they will not have to manually delete the existing entry first.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 commented:
Sorry, the description is not easy to understand - what exactly does this change? Which item or line of text?

Kebap commented:
Select multiple rooms with differing details, open menu to edit properties:
![image](https://user-images.githubusercontent.com/117238/211106983-1c18d221-4213-45bb-908c-aea7ee39be1a.png)

Select any (other) entry from drop-down menu of symbols. 
![image](https://user-images.githubusercontent.com/117238/211107482-f353a499-1a52-4034-9ed6-b08961bf9a56.png)

The text will be selected in this PR. Whereas in PTB it will not be selected.
The idea is, now you can type and override. Whereas in PTB you need to hit a few more keys to remove all the selection.

May I suggest we also select all text when you simply change focus into any of these value boxes: Name, Symbol, Weight.
Whereas currently, you need to select something (else) from the dropdown, this change would make it so simply clicking the box will select all text, so you can start typing to overwrite the content.

vadi2 commented:
I still don't see a difference so I'll leave it to you @Kebap to review.

Kebap reviewed (dismissed):
The existing scope works fine.
Not sure if SlySven wants to enhance it with my suggestion or no.

SlySven commented:
> Not sure if SlySven wants to enhance it with my suggestion or no.

Yeah, I think I can ensure that a field is fully selected on entry  / activation of the dialogue - but after that I don't think I can do anything easily for the Names/Symbol/Weight `QLineEdit`s (the latter two only when there is none or only one value to display) this is because there is no *signal* for the item concerned receiving focus and instead it would have to be done by *focus-in events* I guess and I am a lot less confident I could come up with reliable code to do THAT with out messing up the UI behaviour.

One thing I can/and have nearly finished doing is, given that there are currently nearly half a dozen actions on the context menu that causes the room properties dialogue to show, **to ensure that the focus is directed to the appropriate widget on entry depending on what the user selected instead of always plonking it on the first (the room name) `QLineEdit`**. I have just about completed this and I did hope to push it out yesterday but wasn't able to. I hope to do that later tonight - all being well. :worried: 

*\<aside>This didn't happen because I've been getting random crashes which seemed to be associated with NCQ on a 1TByte drive I installed a couple of months ago. As of yesterday I found out how to disable that, and I thought I had nailed the problem, which it was suggested could be because of my* mature *PC AMD hardware - which I boosted a few weeks back by upgrading from a 1.8GHz to 2.5GHz multicore processor - and large SATA drives with Native Command Queuing. However after juggling file-systems and partitions around to move data off the drive and shoe horn it back onto the other 4 smaller HHDs {thank goodness for LVM :grinning:};  then spending a couple of hours doing a SATA Secure Erase on the drive; then wrestling the drive out from the position which was unfortunately an inaccessible one behind the now larger processor cooler and slotting back in the other one that was temporarily/strategically laying free on its back on top of the DVD-Writer - I've had another hang occurring a short time after getting some atypical "spin-up" whines from one of the remaining drives whilst `grub` was probing for OSes across all the drives in the system. :sob:  I had thought I'd previous tried juggling the connections to the mobo's 6 SATA ports and also switching the SATA data leads but now I am NOT using the either of those that was associated with the prior failures and things seem to be running again... :slightly_smiling_face:\</aside>*

Kebap commented:
> given that there are currently nearly half a dozen actions on the context menu that causes the room properties dialogue to show, to ensure that the focus is directed to the appropriate widget on entry depending on what the user selected instead of always plonking it on the first (the room name)

YAGNI.
These actions will vanish at the earliest convenience. They are simply there to ease transition for this release version only.
No disrespect to your developments honor, but they will have a pretty short shelf life in this case.

SlySven committed:
Revise: select all text in any room properties widget & focus on wanted one

Now the focus is put upon the appropriate widget depending on what the
action that was clicked on the 2D mapper context menu - and all the text
there will be pre-selected so that the first key stroke can remove it ready
for a new entry.

Since the "Lock..." and "Unlock..." actions now both end up in the same
place they are duplicates of each other and can be combined.

For clarity, since I am editing some texts in this area I have taken the
liberty to introduce some numerous texts {i.e. `Qbject::tr(...)` calls with
a third numeric argument (in these cases the number of rooms selected)} so
that the texts can be presented as appropriate for a locale when the number
of rooms that the actions will work on varies.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented:
> These actions will vanish at the earliest convenience. They are simply there to ease transition for this release version only.
> No disrespect to your developments honor, but they will have a pretty short shelf life in this case.

Ah, well,  *dems da breks!*

OTOH By doing them I learnt how to replace *obsolete* `QSignalMapper`s with lambdas - I was aware that it could be done (the Qt documentation drops heavy hints that one needed to do so going forward from around **5.11.x**) but I didn't then grok it. However I can fully grok it now! :nerd_face:  We use it twice by the way elsewhere (they bot have **`// TODO:`** s BTW) once elsewhere in the `T2DMap` class for user menu actions and also in the colour trigger colour selection dialogue. The reason I was considering it here - but learned how to do it with lambdas instead is that I needed the call to `(void) T2DMap::slot_showPropertiesDialog()` to gain an argument to tell it which item to focus on and all of those actions wanted different things but they were all `connect()`ed to that slot - so the secret is to instead get them to `emit` a signal with a different argument value and then arrange for that signal to be connected to the revised slot that takes that argument instead...

SlySven committed:
Merge branch 'development' into Improve_preselectTextInRoomPropertiesWidgets

Kebap commented:
In lieu of upcoming feature freeze I suggest to roll back the last commit and merge the green-light status before.
Then push the new commit to a new branch and we can continue discussion there, as it greatly enlarges the scope.

SlySven commented:
> ...  I suggest to roll back the last commit and merge the green-light status before...

Please identify (by SHA1)  the commit you mean - note that the last commit was a merge to bring in changes from the development branch and it may not be possible to just revert that and still have merge-able code (I haven't yet tried it).

Kebap commented:
https://github.com/Mudlet/Mudlet/pull/6486/commits/449a14369efadcbd697e83f83ef5a062fde948bc

SlySven committed:
Revert: undo/remove feature that moved focus to appropriate control

As requested by @Kebap.

The revisions that adjusted some tool-tips and menu items to include %n
elements or at least have the number provided to the `QObject::tr(...)`
calls to reflect the number of rooms involved have been retained because
otherwise the texts would have had mentions of "room(s)" without any means
for the translators to handle pluralisation in their locale.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Merge branch 'development' into Improve_preselectTextInRoomPropertiesWidgets

SlySven committed:
Revert: undo other changes not germane to original PR subject

Although I think making the translations have a "numerus" form is desirable
it has been deemed that it is not helpful to include it in this PR.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Merge branch 'development' into Improve_preselectTextInRoomPropertiesWidgets

SlySven committed:
Merge branch 'development' into Improve_preselectTextInRoomPropertiesWidgets

Kebap commented:
Stop it bot, you're drunk!
![grafik](https://user-images.githubusercontent.com/117238/216265088-88340a3c-b319-417f-b2df-171a0ee4303e.png)

vadi2 commented:
Yeah... that's a bug, it always assigns a milestone even if one is already applied.

Kebap commented:
> that's a bug

Where do I report upstream?

vadi2 commented:
Just in this PR, this workflow does it - https://github.com/Mudlet/Mudlet/blob/development/.github/workflows/tag-pull-requests.yml

Should be updated to check if a milestone was already assigned

SlySven committed:
Merge branch 'development' into Improve_preselectTextInRoomPropertiesWidgets
